android/src/main/java/com/twiliovoicereactnative/NotificationUtility.java

```diff
@@ -35,6 +33,50 @@
 
 class NotificationUtility {
   private static final SecureRandom secureRandom = new SecureRandom();
+  private static class NotificationResource {
+    private final Context ctx;
+    private final String smallIconId;
+    private final String contentTextId;
+    public NotificationResource(Context context, String smallIconId, String contentTextId) {
+      ctx = context;
+      this.smallIconId = smallIconId;
+      this.contentTextId = contentTextId;
+    }
+    int getSmallIconId() {
+      return getDrawableResourceId(smallIconId);
+    }
+    public String getContent(CallInvite callInvite) {
+      final String from = (null != callInvite) ? getDisplayName(callInvite) : "";
+      return ctx.getString(getTextResourceId(contentTextId))
+        .replaceAll("\\[%from\\]", from);
+    }
+    public String getContent(String to) {
+      return ctx.getString(getTextResourceId(contentTextId))
+        .replaceAll("\\[%to\\]", to);
+    }
+
+    @SuppressLint("DiscouragedApi")
+    private int getDrawableResourceId(String id) {
+      return ctx.getResources().getIdentifier(id, "drawable", ctx.getPackageName());
+    }
+    @SuppressLint("DiscouragedApi")
+    private int getTextResourceId(String id) {
+      return ctx.getResources().getIdentifier(id, "string", ctx.getPackageName());
+    }
+    private static String getDisplayName(@NonNull CallInvite callInvite) {
+      final String title = callInvite.getFrom();
+      Map<String, String> customParameters = callInvite.getCustomParameters();
+      // If "displayName" is passed as a custom parameter in the TwiML application,
+      // it will be used as the caller name.
```

Comment:
I'm not sure if this is well documented anywhere, we should sync up on this and check our Wagtail documentation.